Detectors/TPC/reconstruction/src/KrBoxClusterFinder.cxx

```diff
@@ -112,7 +86,53 @@ void KrBoxClusterFinder::fillADCValue(int cru, int rowInSector, int padInRow, in
     adcValue /= correctionFactor;
   }
 
-  mMapOfAllDigits[timeBin][rowInSector][corPad] = adcValue;
+  mTempTimeSlice[rowInSector][corPad] = adcValue;
+}
+
+void KrBoxClusterFinder::fillTempTimeSlice(std::vector<o2::tpc::Digit>& eventSector, const int timeSlice) {
+  // No third argument needed since iterator gets popped
+  for (std::vector<o2::tpc::Digit>::iterator digit = eventSector.begin(); digit != eventSector.end();) {
+    const int time = digit->getTimeStamp();
+    if(time != timeSlice) {
+      return;
+    }
+
+    const int cru = digit->getCRU();
+    mSector = cru / CRU::CRUperSector;
+
+    const int rowInSector = digit->getRow();
+    const int padInRow    = digit->getPad();
+    float adcValue        = digit->getChargeFloat();
+
+    fillADCValueInTempSlice(cru, rowInSector, padInRow, adcValue);
+
+    // We don't need this digit anymore:
+    digit = eventSector.erase(digit);
+  }
+  return;
+}
+
+const std::vector<KrCluster>& KrBoxClusterFinder::loopOverSector(std::vector<o2::tpc::Digit>& eventSector) {
+  mClusters.clear();
+  createInitialMap(eventSector);
+  for(int iTimeSlice = mMaxClusterSizeTime; iTimeSlice < MaxTimes - mMaxClusterSizeTime; ++iTimeSlice) {
+    findLocalMaxima(true, iTimeSlice);
+    popFirstTimeSliceFromMap();
+    fillTempTimeSlice(eventSector, iTimeSlice);
+    addTimeSliceToMap();
+    clearTempTimeSlice();
+  }
+  return mClusters;
+}
+
+void KrBoxClusterFinder::resetADCMap() {
+  // Has to be reimplemented!
+  return;
+}
+
+void KrBoxClusterFinder::fillADCValue(int cru, int rowInSector, int padInRow, int timeBin, float adcValue) {
+  // Has to be reimplemented!
+  return;
```

Comment:
These functions are needed for the workflow but do not work anymore as they were implemented.

Reply:
I think we will need to implement it in a different way. I already have an idea.

Reply:
Very cool, thanks. Do you want to explain your ideas?

Reply:
The DPL device currently fills the data digit by digit. We will now first need to buffer all digits and call `loopOverSector` instead of `fillADCValue`. I can quickly implement it once the changes are committed.